.github/workflows/ci.yml

```diff
@@ -179,3 +181,47 @@ jobs:
         run: |
           cd $BUILD_DIR/scripts/code_generation/fakegen
           valgrind --tool=memcheck --leak-check=yes ./fakegen_test
+  
+  generate-dbc:
+      name: Generate CAN DBC file
+      runs-on: ubuntu-22.04
+      container:
+        image: ubcformulaelectric/environment:latest
+        credentials:
+          username: ubcformulaelectric
+          password: ${{ secrets.docker_token }}
+      steps:
+        - name: Checkout repository
+          uses: actions/checkout@v2
+          with: # TODO check if submodules are required (can be optimized away) (From bin-dbc-gen)
+            submodules: recursive
+        - name: Generate DBC
+          run: |
+            python3 scripts/code_generation/jsoncan/generate_can_from_json.py --only_dbc --can_data_dir "can_bus/json" --dbc_output "can_bus/dbcs/CanMsgs.dbc"
+        - name: Upload DBC to Artifact
+          uses: actions/upload-artifact@v2
+          with:
+            name: CanMsgs.dbc
+            path: ./can_bus/dbcs/CanMsgs.dbc
+
+  make-release:
+    name: Add DBC to Pull_Request-specific release
+    runs-on: ubuntu-22.04
+    needs: [test-fakegen, test-shared, test-thruna, build-firmware-thruna, codegen, generate-dbc]
+    steps:
+      - name: Extract branch name
+        shell: bash
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+        id: extract_branch
+      - name: Download DBC from artifact
+        uses: actions/download-artifact@v2
+      - name: Create Release
+        uses: "marvinpinto/action-automatic-releases@latest"
+        with:
+          repo_token: "${{ secrets.GITHUB_TOKEN }}"
+          automatic_release_tag: "${{steps.extract_branch.outputs.branch}}"
+          title: "DBC Generation"
+          tag: ${{ github.sha }}
```

Comment:
Something between these commands is causing the tag to attach to the commit merging into master. This is not intended behaviour. Indended is that the tag attaches to the latest commit in the branch this workflow is running on.

Reply:
my suspicion is that on immediate pr, the tag is linked to that specific pr. However any future commits cause the tag to be tossed around.